main.go

```diff
@@ -213,15 +221,111 @@ func fetchOrCreateTestPlan(ctx context.Context, cfg config.Config, files []strin
 		}
 		// Create the fallback plan
 		fmt.Println("Could not fetch plan from server, using fallback mode. Your build may take longer than usual.")
-		testPlan = plan.CreateFallbackPlan(testCases, cfg.Parallelism)
+		testPlan = plan.CreateFallbackPlan(files, cfg.Parallelism)
 	}
 
 	// The server can return an "error" plan indicated by an empty task list (i.e. `{"tasks": {}}`).
 	// In this case, we should create a fallback plan.
 	if len(testPlan.Tasks) == 0 {
 		fmt.Println("Test splitter server returned an error, using fallback mode. Your build may take longer than usual.")
-		testPlan = plan.CreateFallbackPlan(testCases, cfg.Parallelism)
+		testPlan = plan.CreateFallbackPlan(files, cfg.Parallelism)
 	}
 
 	return testPlan, nil
 }
+
+type fileTiming struct {
+	Path     string
+	Duration time.Duration
+}
+
+// createRequestParam creates the request parameters for the test plan.
+// If SplitByExample is disabled (default), it will return the default params that contain all the files.
+// If SplitByExample is enabled, it will split the slow files into examples and return it along with the rest of the files.
+//
+// Error is returned if there is a failure to fetch test file timings or to get the test examples from test files when SplitByExample is enabled.
+func createRequestParam(cfg config.Config, files []string, client api.Client, runner TestRunner) (api.TestPlanParams, error) {
+	if !cfg.SplitByExample {
+		debug.Println("Splitting by file")
```

Comment:
Does your debug package print the time as well as the log line?

Reply:
Currently, it will print the time and debug message. Do you think we need to add the line number as well?

It looks like this currently
```
2024/06/17 10:04:05 DEBUG: Fetching test plan
2024/06/17 10:04:05 DEBUG: No test plan found, creating a new plan
2024/06/17 10:04:05 DEBUG: Splitting by example
2024/06/17 10:04:05 DEBUG: Fetching timings for 1748 files
...
```

Reply:
Nope I was just checking that we had timestamps!